### ISSUE TITLE
witness_node gets stuck

### ISSUE THREAD

**Issue by pmconrad** [cryptonomex/graphene#478]
My witness_node got stuck two days ago two hours after it had been restarted without --replay-blockchain. I didn't care much because I was about to upgrade to the latest version anyway.

Today both my witness_node and my backup node got stuck at the same block. Both were running the latest version, both had been restarted after the last reindex. The last line it printed on stdout was

```
3165295ms th_a       application.cpp:496           handle_block         ] Got block: #1712872 time: 2015-12-12T10:52:45 latency: 296 ms from: elmato  irreversible: 1712854 (-18)
```

(The first time it got stuck it happened at block 1655316.)

The logs/p2p/p2p.log shows that it continued to receive blocks from the network for about 30 seconds, but apparently these weren't processed anymore. After that, p2p.log contains regular reports about 20 connected clients, but no apparent communication.

**Comment by pch957** [comment]
maybe it's relative
https://bitsharestalk.org/index.php/topic,20631.msg266292.html#msg266292

```
(gdb) bt
#0  0x0000000000dcecf5 in void fc::divide<fc::uint128>(fc::uint128 const&, fc::uint128 const&, fc::uint128&, fc::uint128&) ()
#1  0x0000000000dcedf4 in fc::uint128::operator/=(fc::uint128 const&) ()
#2  0x0000000000be6d9a in graphene::chain::database::clear_expired_orders() ()
#3  0x0000000000beda0f in graphene::chain::database::_apply_block(graphene::chain::signed_block const&) ()
#4  0x0000000000bee835 in graphene::chain::database::apply_block(graphene::chain::signed_block const&, unsigned int) ()
#5  0x0000000000beed34 in graphene::chain::database::_push_block(graphene::chain::signed_block const&) ()
#6  0x0000000000bf0af3 in graphene::chain::database::push_block(graphene::chain::signed_block const&, unsigned int) ()
#7  0x0000000000a3783d in graphene::app::detail::application_impl::handle_block(graphene::net::block_message const&, bool, std::vector<fc::ripemd160, std::allocator<fc::ripemd160> >&) ()
#8  0x0000000000f3f9d7 in fc::detail::functor_run<graphene::net::detail::statistics_gathering_node_delegate_wrapper::handle_block(graphene::net::block_message const&, bool, std::vector<fc::ripemd160, std::allocator<fc::ripemd160> >&)::{lambda()#1}>::run(void*, fc::detail::functor_run<graphene::net::detail::statistics_gathering_node_delegate_wrapper::handle_block(graphene::net::block_message const&, bool, std::vector<fc::ripemd160, std::allocator<fc::ripemd160> >&)::{lambda()#1}>) ()
#9  0x0000000000dedfa4 in fc::task_base::run_impl() ()
#10 0x0000000000debbaf in fc::thread_d::process_tasks() ()
#11 0x0000000000debdf1 in fc::thread_d::start_process_tasks(long) ()
#12 0x0000000001061621 in make_fcontext ()
#13 0x0000000000000000 in ?? ()
```

**Comment by SpectralVision** [comment]
Something similar happened to me today. Updated to 2.0.151215 last night, and my witness got stuck today, after less than 12 hours.

The first missing block was reported at 12:59 UTC. A portion of the p2p log file from 12:57 UTC and onwards is available at the following link:

https://drive.google.com/file/d/0BzEEZBS7b6xPb2c2bGxfeVQ4Y0k/view?usp=sharing

**Comment by pmconrad** [comment]
Interesting. I updated two nodes last night, and both got stuck at
Got block: #1829900 time: 2015-12-16T12:59:36
Both were restarted since the last rescan. So far I have only seen these hangs after restarting without rescanning.

**Comment by bytemaster** [comment]
I have seen this issue and am looking for the cause.  It is interesting that it just started happening recently.

**Comment by pch957** [comment]
it happened when settlement expired
https://bitsharestalk.org/index.php/topic,20631.msg266820.html#msg266820

**Comment by SpectralVision** [comment]
I have some more information that may be useful, a screen output from my witness and GDB while it is still stuck. I stepped a bit forward, might be helpful to someone familiar with the code.

[2015-12-17-gdb.txt](https://github.com/cryptonomex/graphene/files/65836/2015-12-17-gdb.txt)

The P2P-log from about 5min before the crash is available here:
https://drive.google.com/file/d/0BzEEZBS7b6xPbzducHhvOXpmdWc/view?usp=sharing

Extra information that may or may not be relevant, but I will put it up here anyway just in case:

I can add that for me this crash started happening about the same time as I started using a new version of xeroc's price feed script. However, that script is not running on my seed node, which is also getting stuck.

Also, the first time this crash happened to me, two of my VPS'es crashed at the same time, and I was unable to connect back to them for an extended period of time, even when doing hard reset from the web interface. That has never happened before or again.

Yesterday, in addition to updating the node to version 16b, I also noticed that my /boot partition was 99% full on my witness node, and corrected that. The seed node was "only" 86% full, and that has also crashed, so it may not be relevant.
